apps/browser/qml/pages/components/PopUpMenuItem.qml

```diff
@@ -0,0 +1,215 @@
+                iconSource: "image://theme/icon-m-tab-new"
+                //% "New tab"
+                text: qsTrId("sailfish_browser-la-new_tab")
+                onClicked: {
+                    webView.privateMode = false
+                    overlay.toolBar.enterNewTabUrl()
+                }
+            }
+
+            OverlayListItem {
+                height: Theme.itemSizeSmall
+                iconWidth: root.iconWidth
+                horizontalOffset: root.horizontalOffset
+                iconSource: "image://theme/icon-m-incognito-new"
+                //% "New private tab"
+                text: qsTrId("sailfish_browser-la-new_private_tab")
+                onClicked: {
+                    webView.privateMode = true
+                    overlay.toolBar.enterNewTabUrl()
+                }
+            }
+        }
+
+        Column {
+            width: parent.width
+
+            OverlayListItem {
+                height: Theme.itemSizeSmall
+                iconWidth: root.iconWidth
+                horizontalOffset: root.horizontalOffset
+                iconSource: "image://theme/icon-m-search-on-page"
+                enabled: webView.contentItem
+                //% "Search on page"
+                text: qsTrId("sailfish_browser-la-search_on_page")
+
+                onClicked: {
+                    overlay.toolBar.findInPageActive = true
+                    overlay.toolBar.findInPage()
+                }
+            }
+
+            OverlayListItem {
+                height: Theme.itemSizeSmall
+                iconWidth: root.iconWidth
+                horizontalOffset: root.horizontalOffset
+                iconSource: "image://theme/icon-m-add-to-grid"
+                //% "Add to apps grid"
+                text: qsTrId("sailfish_browser-la-add_to_apps_grid")
+                onClicked: pageStack.animatorPush("AddToAppGridDialog.qml",
+                                                  {
+                                                      "url": url,
+                                                      "title": title,
+                                                      "desktopBookmarkWriter": desktopBookmarkWriter,
+                                                      "bookmarkWriterParent": pageStack
+                                                  })
+            }
+
+            OverlayListItem {
+                height: Theme.itemSizeSmall
+                iconWidth: root.iconWidth
+                horizontalOffset: root.horizontalOffset
+                iconSource: "image://theme/icon-m-file-download-as-pdf"
+                //% "Save web page as PDF"
+                text: qsTrId("sailfish_browser-la-save_as-pdf")
+
+                onClicked: {
+                    if (DownloadManager.pdfPrinting) {
+                        pdfPrintingNotice.show()
+                    } else {
+                        overlay.toolBar.savePageAsPDF()
+                    }
```

Comment:
Perhaps selecting this item should also close the menu, to make it feel more responsive.

Reply:
done

Reply:
Thank you!